java/console/src/main/java/Main.java

```diff
@@ -0,0 +1,267 @@
+    private static final String AUDIO_RESPONSE_WAV_FILE = "audio_response.wav";
+
+    public static void main(String[] args) {
+        AudioFile audioFile = new AudioFile(Paths.get("whats_the_weather_pcm16_24khz_mono.wav"))
+                .setBytesPerSample(16)
+                .setSampleRate(24000);
+
+        RealtimeAsyncClient client = buildClient(true);
+
+        // We have a composite disposable to collect our subscriptions
+        Disposable.Composite disposables = Disposables.composite();
+
+        // Set up event consumers for our server events of interest:
+        // - SessionCreatedEvent
+        // - ResponseAudioDeltaEvent
+        // - ResponseAudioTranscriptDeltaEvent
+        disposables.addAll(Arrays.asList(
+                client.getServerEvents()
+                        .takeUntil(serverEvent -> serverEvent instanceof SessionCreatedEvent)
+                        .ofType(SessionCreatedEvent.class)
+                        .subscribe(Main::consumeSessionCreated, Main::consumeError, Main::onSessionCreatedCompleted),
+                client.getServerEvents()
+                        .takeUntil(serverEvent -> serverEvent instanceof ResponseAudioDoneEvent)
+                        .ofType(ResponseAudioDeltaEvent.class)
+                        .subscribe(Main::consumeAudioDelta, Main::consumeError, Main::onAudioResponseCompleted),
+                client.getServerEvents()
+                        .takeUntil(serverEvent -> serverEvent instanceof ResponseAudioTranscriptDoneEvent)
+                        .ofType(ResponseAudioTranscriptDeltaEvent.class)
+                        .subscribe(Main::consumeAudioTranscriptDelta, Main::consumeError,
+                                Main::onAudioResponseTranscriptCompleted)));
+
+        // Initializing connection to server
+        client.start().block();
+
+        // Configure the realtime session
+        client.sendMessage(new SessionUpdateEvent(
+                new RealtimeRequestSession()
+                        .setVoice(RealtimeVoice.ALLOY)
+                        .setTurnDetection(
+                                new RealtimeServerVadTurnDetection()
+                                        .setThreshold(0.5)
+                                        .setPrefixPaddingMs(300)
+                                        .setSilenceDurationMs(200))
+                        .setInputAudioTranscription(new RealtimeAudioInputTranscriptionSettings(
+                                RealtimeAudioInputTranscriptionModel.WHISPER_1))
+                        .setModalities(Arrays.asList(RealtimeRequestSessionModality.AUDIO,
+                                RealtimeRequestSessionModality.TEXT))))
+                .block();
+
+        // Send audio file with our prompt
+        sendAudioFileAsync(client, audioFile).block();
+
+        try {
+            // We await for subscriptions that we didn't block on
+            Thread.sleep(10000);
```

Comment:
This is something I am not entirely happy with in the original samples I wrote. To give you an alternative to sleeping the thread, you could block on getting a `ResponseDoneEvent`. I did that in the [ToolCall sample](https://github.com/Azure/azure-sdk-for-java/pull/43336/files#diff-d14681f02eaba0ef25d4712c9aa3bd2070ce957e7701f57d5d6d3ffd53228056R90-R92) where I know ahead of the interaction with the service that I will get 2 `ResponseDoneEvent`s (one for the tool call and one with the reply to the prompt).

Here if you are expecting a single one you could add instead:

```java
 client.getServerEvents().ofType(ResponseDoneEvent.class).single().block();
// or, equivalently
 client.getServerEvents().ofType(ResponseDoneEvent.class).take(1).blockLast();
```

Whichever option you find more readable "sample-wise"